exodus/restful_api/tests.py

```diff
@@ -8,63 +8,147 @@
 from reports.models import Application, Report, Permission, Apk
 from trackers.models import Tracker
 
-
 DUMMY_HANDLE = 'com.example'
-API_HANDLE_DETAILS_PATH = '/api/search/{}/details'.format(DUMMY_HANDLE)
-API_TRACKERS_PATH = '/api/trackers'
-API_APPLICATIONS_PATH = '/api/applications'
 
 
-class RestfulApiGetAllApplicationsTests(APITestCase):
+class RestfulApiApplicationTests(APITestCase):
+    PATH = '/api/applications'
 
-    def force_authentication(self):
+    def _force_authentication(self):
         user = User.objects.create_user('username', 'Pas$w0rd')
         self.client.force_authenticate(user)
 
+    def test_returns_unauthorized_when_no_auth(self):
+        response = self.client.get(self.PATH)
+        self.assertEqual(response.status_code, 401)
+
     def test_returns_empty_json_when_no_applications(self):
-        self.force_authentication()
-        response = self.client.get(API_APPLICATIONS_PATH)
+        self._force_authentication()
+        response = self.client.get(self.PATH)
 
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.json()['applications'], [])
 
     def test_returns_applications_with_report_last_update(self):
-        self.force_authentication()
-        report = Report.objects.create()
+        self._force_authentication()
+        report = Report.objects.create(id=1234)
         application = Application.objects.create(
             name='app_name',
             handle='handle',
             report=report
         )
 
-        response = self.client.get(API_APPLICATIONS_PATH)
+        expected_json = {
+            'applications': [
+                {
+                    'id': application.id,
+                    "handle": application.handle,
+                    "name": application.name,
+                    "creator": "",
+                    "downloads": "",
+                    "app_uid": "",
+                    "icon_phash": "",
+                    "report_updated_at": report.updated_at.timestamp()
+                },
+            ]
+        }
+
+        response = self.client.get(self.PATH)
 
         self.assertEqual(response.status_code, 200)
-        self.assertContains(response, application.name, 1)
+        self.assertEqual(response.json(), expected_json)
+
+
+class RestfulApiSearchHandleDetailsTests(APITestCase):
+    PATH = '/api/search/{}/details'.format(DUMMY_HANDLE)
 
-        report_updated_at = application.report.updated_at
-        response_application = response.json()['applications'][0]
+    def _force_authentication(self):
+        user = User.objects.create_user('username', 'Pas$w0rd')
```

Comment:
I think I don't understand this line, but it seem strange to me to see default username and password harcoded, is it normal ?

Reply:
This is to test the authentication on the API (see https://www.django-rest-framework.org/api-guide/testing/)
We need to create a user in the test database which is created just for the tests and deleted afterwards. The password does not matter.